generate/pki.rb

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ruby
+
+require 'yaml'
+require 'fileutils'
+
+SCRIPT_DIR = File.expand_path(File.dirname(__FILE__))
+pki_to_generate = YAML.load_file(ARGV[0] ||= "#{SCRIPT_DIR}/pki.yml")
+pki_dir = 'pki'
+FileUtils::mkdir_p pki_dir
+Dir.chdir pki_dir
+
+def generate_cert(type, common_name, file, ca_cert, ca_key)
+    command = <<-COMMAND
+      sed 's/$COMMON_NAME/#{common_name}/' #{SCRIPT_DIR}/template-csr.json | \
+      cfssl gencert -config #{SCRIPT_DIR}/cfssl-config.json -profile "#{type}" -ca #{ca_cert} -ca-key #{ca_key} /dev/stdin | \
+      cfssljson -bare #{file}
+    COMMAND
+    `#{command}`
```

Comment:
error checking?

also use `%x` instead of this interpolation

I.E.

```
require 'English'

...

output = %x{
  sed 's/$COMMON_NAME/#{common_name}' #{SCRIPT_DIR}/template-csr.json
  | cfssl gencert -config #{SCRIPT_DIR}/cfssl-config.json
                  -profile "#{type}"
                  -ca #{ca_cert}
                  -ca-key #{ca_key} /dev/stdin
  | cfssljson -bare #{file}
}
raise output unless $CHILD_STATUS.success?
```

Ideally you wouldn't do this though